packages/api/amplify_api/test/graphql_helpers_test.dart

```diff
@@ -40,8 +41,118 @@ void main() {
+
+      expect(req.document, expectedDoc);
+      expect(_deepEquals(req.variables, {'id': id}), isTrue);
+      expect(req.modelType, FirstMtmRelation.classType);
+      expect(req.decodePath, 'getFirstMtmRelation');
+    });
+
+    test(
+        'ModelMutations.create() on mtmRelationModel should have both relationships',
+        () {
+      const expectedDoc = 'mutation createFirstMtmRelation('
+          '\$input: CreateFirstMtmRelationInput!, '
+          '\$condition:  ModelFirstMtmRelationConditionInput) { '
+          'createFirstMtmRelation(input: \$input, condition: \$condition) '
+          '{ $mtmRelationSelectionSet } }';
+      final expectedVars = {
+        'input': {
+          'id': 'firstMtmRelationId',
+          'manyToManyPrimaryID': 'mtmPrimaryId',
+          'manyToManySecondaryID': 'mtmSecondaryId'
+        }
+      };
+
+      ManyToManyPrimary primary = ManyToManyPrimary(
+        id: 'mtmPrimaryId',
+        name: 'first',
+      );
+      ManyToManySecondary secondary =
+          ManyToManySecondary(id: 'mtmSecondaryId', name: 'second');
+
+      FirstMtmRelation relation = FirstMtmRelation(
+          id: 'firstMtmRelationId',
+          manyToManyPrimary: primary,
+          manyToManySecondary: secondary);
+      GraphQLRequest<FirstMtmRelation> req =
+          ModelMutations.create<FirstMtmRelation>(relation);
+
+      expect(req.document, expectedDoc);
+      expect(_deepEquals(req.variables, expectedVars), isTrue);
+      expect(req.modelType, FirstMtmRelation.classType);
+      expect(req.decodePath, 'createFirstMtmRelation');
+    });
+
+    test(
+        'ModelMutations.update() on mtmRelationModel should have both relationships',
+        () {
+      const expectedDoc = 'mutation updateFirstMtmRelation('
+          '\$input: UpdateFirstMtmRelationInput!, '
+          '\$condition:  ModelFirstMtmRelationConditionInput) { '
+          'updateFirstMtmRelation(input: \$input, condition: \$condition) '
+          '{ $mtmRelationSelectionSet } }';
+      final expectedVars = {
+        'input': {
+          'id': 'firstMtmRelationId',
+          'manyToManyPrimaryID': 'mtmPrimaryId',
+          'manyToManySecondaryID': 'mtmSecondaryId'
+        },
+        'condition': null
+      };
+
+      ManyToManyPrimary primary = ManyToManyPrimary(
+        id: 'mtmPrimaryId',
+        name: 'first',
+      );
+      ManyToManySecondary secondary =
+          ManyToManySecondary(id: 'mtmSecondaryId', name: 'second');
+
+      FirstMtmRelation relation = FirstMtmRelation(
+          id: 'firstMtmRelationId',
+          manyToManyPrimary: primary,
+          manyToManySecondary: secondary);
+      GraphQLRequest<FirstMtmRelation> req =
+          ModelMutations.update<FirstMtmRelation>(relation);
+
+      expect(req.document, expectedDoc);
+      expect(_deepEquals(req.variables, expectedVars), isTrue);
+      expect(req.modelType, FirstMtmRelation.classType);
+      expect(req.decodePath, 'updateFirstMtmRelation');
```

Comment:
request: Could you please add a test that the decoder handles this situation as expected? See below examples with mocked server response. Will need a mocked server response (dart map) a like AppSync would provide in this scenario passed directly to decoder and that resulting response has model with relationships populated as expected.